### DISCUSSION TITLE
how to create stream from modbus device

### DISCUSSION THREAD

**Discussion by zero530** [lf-edge/ekuiper#1320] [General]
Dear all,

we need to query data from the modbus device and filter the data (e.g: Temperature>30) and send it to mqtt.

What can I do? thanks

**Comment by ngjaying** [comment]
We do not connect to modbus device directly, you'll need another application like neuron/edgex to fetch data from it. eKuiper already supports to conenct to [neuron](https://neugates.io/) or [edgex](https://www.edgexfoundry.org/). Check the tutorials https://ekuiper.org/docs/en/latest/tutorials/neuron/neuron_integration_tutorial.html or https://ekuiper.org/docs/en/latest/edgex/edgex_rule_engine_tutorial.htm.

If you don't want to introduce other product, you can also create a source plugin for eKuiper to ingest the data from modbus device.
